Application/src/Parser.java

```diff
@@ -0,0 +1,135 @@
+import Model.diagramObject;
+import javafx.scene.control.Alert;
+import com.google.gson.Gson;
+
+
+/**
+ * @author Rashad Kamsheh & Isabelle Törnqvist
+ * @since 2017-10-16
+ *
+ * made with usage of Gson
+ * Gson is released under the Apache 2.0 license.
+
+    Copyright 2008 Google Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+public class Parser {
+
+    /** defining two different counter variables that will increment each time the diagram calling method
+    *that they are is called in order to differentiate between the imported diagrams
+    */
+    // odd number for the First Diagram counter
+    static int counter = 1;
+    // even number for the parallel Diagram counter
+    static int parallelCounter = 2;
+
+    static String processes = "";
+    static String properProcesses = "";
+
+    static String classNames = "";
+    static String properClassNames = "";
+
+    static String firstDiagram = "";
+    static String properFirstDiagram = "";
+
+    static String parallelDiagram = "";
+    static String properParallelDiagram = "";
+
+
+    /**
+     * Parses the imported JSON files
+     * @param inputJSON the imported collection of Strings that make up the JSON files
+     */
+
+    public static void parse(String inputJSON) {
+        Gson gson = new Gson();
+
+        try {
+            diagramObject parsedDiagram;
```

Comment:
What does this line do? Variable name variable name?

Reply:
diagramObject = object, parsedDiagram = variable name. See diagramObject.java
Shouldn't need to clarify this in the code (?), but good that you asked

Reply:
Shouldnt it be DiagramObject then?